src/mscorlib/src/System/Collections/Generic/List.cs

```diff
@@ -1051,6 +1054,29 @@ public void Sort(IComparer<T> comparer)
             return true;
         }
 
+        private class PooledIEnumerator : IEnumerator<T>
+        {
+            private Enumerator _enumerator;
+
+            internal PooledIEnumerator Initalize(List<T> list)
+            {
+                _enumerator = new Enumerator(list);
+                return this;
+            }
+            public bool MoveNext() => _enumerator.MoveNext();
+            public T Current => _enumerator.Current;
+            object IEnumerator.Current => Current;
+            public void Dispose()
+            {
+                _enumerator = default(Enumerator);
```

Comment:
I don't think it's common but it certainly is a behavior change that needs to be considered.